main/managementinterface.go

```diff
@@ -819,6 +820,39 @@ func handleUpdatePostRequest(w http.ResponseWriter, r *http.Request) {
 	go delayReboot()
 }
 
+// Upload an update file for Pong
+func handlePongUpdatePostRequest(w http.ResponseWriter, r *http.Request) {
+	setNoCache(w)
+	setJSONHeaders(w)
+	overlayctl("unlock")
```

Comment:
I think you can remove the overlay unlock call. You only seem to be writing to /tmp anyway?
If not, please don't forget to lock again when done.